support/k8s/k8s-workload-registrar/mode-crd/controllers/pod_controller.go

```diff
@@ -316,3 +364,33 @@ func (r *PodReconciler) generateSpiffeIDPath(pod *corev1.Pod) (string, error) {
 	}
 	return spiffeIDPathBuilder.String(), nil
 }
+
+func (r *PodReconciler) createDNSNames(pod *corev1.Pod) ([]string, error) {
+	dnsNames := make([]string, 0, len(r.c.DNSNameTemplates))
+	// Create the IdentityMaps struct, with Pod and Context map
+	templateMaps := IdentityMaps{
+		Context: r.c.Context,
+		Pod: PodInfo{
+			ServiceAccount: pod.Spec.ServiceAccountName,
+			Namespace:      pod.Namespace,
+			Name:           pod.Name,
+			UID:            pod.UID,
+			Hostname:       pod.Spec.Hostname,
+			NodeName:       pod.Spec.NodeName,
+		},
+	}
+
+	for i, dnsNameTemplate := range r.dnsNameTemplates {
+		var dnsNameBuilder strings.Builder
+		if err := dnsNameTemplate.Execute(&dnsNameBuilder, templateMaps); err != nil {
+			r.c.Log.WithError(err).WithFields(logrus.Fields{
+				"dns_name_template":      r.c.DNSNameTemplates[i],
+				"dns_name_template_maps": templateMaps,
+			}).Error("Error executing the dns name template")
+			return nil, err
```

Comment:
In what cases might this fail? Do we want to fail the entire SVID here or just skip over the bad name?

Reply:
It may not ever fail in this case since we've already done the initial validation when creating the reconciler. I can just `continue` if that would be preferred (still log an error though).

Reply:
I thought a little more about it and feel that perhaps it's better to just fail entirely, the signal is stronger that way.